### PR TITLE
KAFKA-2724: ZK Auth documentation.

### PR DESCRIPTION
**Pull request by fpj**

### CODE REVIEW

**Comment by fpj** [comment]
The editor I used added a lot of metadata, I'm working on removing this, but the text itself should be ok.

File: docs/security.html [hunk 1]
```diff
@@ -72,7 +73,7 @@
             <u>Important notes:</u>
             <ol>
                 <li>KafkaServer is a section name in JAAS file used by KafkaServer/Broker. This section tells Kafka Server which principal to use and which keytab this principal is stored. It allows Kafka Server to login using the keytab specified in this section.</li>
-                <li>Client section is used to authenticate a SASL connection with zookeeper. It also allows a broker to set SASL ACL on zookeeper nodes which locks these nodes down so that only kafka broker can modify. It is important to have the same principal name across all the brokers.</li>
+                <li>Client section is used to authenticate a SASL connection with zookeeper. It also allows a broker to set SASL ACL on zookeeper nodes which locks these nodes down so that only kafka broker can modify. It is necessary to have the same principal name across all the brokers. If you want to use a section name other than Client, 2. then you need to set the system property <tt>zookeeper.sasl.client</tt> to the appropriate name (<i>e.g.</i>, <tt>-Dzookeeper.sasl.client=ZkClient</tt>).</li>
```
Comment (gwenshap) [inline docs/security.html]: "Client, 2. then" -> "Client, then"

File: docs/security.html [hunk 2]
```diff
@@ -114,4 +115,42 @@
...
+To enable ZooKeeper authentication on brokers, there are two necessary steps:
+<ol>
+	<li> Create a JAAS login file and set the appropriate system property to point to it as described above</li>
+	<li> Set the configuration property <tt>zookeeper.set.acl</tt> in each broker to true</li>
+</ol>
+
+The metadata stored in ZooKeeper is such that only brokers will be able to modify the corresponding znodes, but znodes are world readable. The rationale behind this decision is that the data stored in ZooKeeper is not sensitive, but inappropriate manipulation of znodes can cause cluster disruption.
+
+<h4><a id="zk_authz_migration">7.5.2 Migrating clusters</a></h4>
+If you have a cluster running an older version of simply with security disabled, and you want to make the cluster secure, then for ZooKeeper authentication, you need to execute the following steps in this order:
```
Comment (gwenshap) [inline docs/security.html]: "older version of simply" -> "older version of Zookeeper"? Or "older version of Kafka"?

**Comment by gwenshap** [comment]
Two minor comments and LGTM!

Great explanation of both rationale and actions.

**Pull request closed without merging**